### PR TITLE
Show movies from watch list in person detail page

### LINKED ISSUE #543: Show movies from both watch history and watch list in person detail page
### Description

Let's say I want to watch a movie from my watch list, but I can't decide what to watch. What I do is prioritize watching movies starring my favorite actor. If I'm lucky, I can remember which movie my favorite actor is in or I can see it from the movie cover. But in cases where I'm unlucky, I have to open the movies on my watch list and check who's starring which is very tedious (considering that I have to click 2 times in order to go to movie detail page in `/watchlist` page).

### Desired Behavior

When we go to the actor's page, we see which of the movies from our watch history he starred in. Similarly, it would be really nice if it could be seen which of the movies from our watch list he is starring. This way, I can choose an actor and when I went to his page, I can easily see his movies that I had not seen before and were on my watch list.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by sahinakkaya**
Fixes #543 

I mostly copied existing code and adjusted it to show movies from watch list. I tested it for my user and it works for me.

### CODE REVIEW

**Comment by sahinakkaya** [comment]
After seeing this in action, I think it might be a good idea to also show the same information for directors. Let me know what you think.

**Comment by sahinakkaya** [comment]
Some screenshots:
resim


resim

**Comment by leepeuker** [comment]
What about showing all movies together (losing the "From your watchlist") and shade the posters of movies not yet watched or something like that. If we add e.g. general playlists later we probably want to include movies which are on an users playlist here too.

The whole person page is a construction site, so all ideas are welcome

**Comment by leepeuker** [comment]
What about something like this. We could reuse the filter logic from the other sites and edit it so that you can select to e.g. only show watched, on watch list, etc or select how the movies should be ordered

image

**Comment by sahinakkaya** [comment]
Yeah, these ideas sound good too. The bad news is I don't know if can implement them :D (sure I can but I mean if I can do it in a reasonable amount of time). 

Btw, the whole point of opening this pull request for me is to be able to easily see the movies from watch list at first glance. So I am not sure how we can make this possible with filtering. Oh... What about combining both of your ideas? Show a filter box, allow users to filter movies based on them being in watch list, history or other custom lists AND shade the posters of movies not yet watched?

File: src/Domain/Movie/MovieRepository.php [hunk 1]
```diff
@@ -898,6 +898,42 @@ public function fetchWithDirector(int $personId, int $userId) : array
...
+    {
+        return $this->dbConnection->fetchAllAssociative(
+            <<<SQL
+            SELECT DISTINCT m.*, mur.rating as userRating
+            FROM movie m
+            JOIN movie_cast mc ON m.id = mc.movie_id
+            JOIN person p ON mc.person_id = p.id
+            JOIN watchlist wl ON m.id = wl.movie_id
+            LEFT JOIN movie_user_rating mur ON wl.movie_id = mur.movie_id and mur.user_id = ?
+            WHERE p.id = ? AND m.id IN (SELECT DISTINCT movie_id FROM watchlist) AND wl.user_id = ?
```
Comment (leepeuker) [inline src/Domain/Movie/MovieRepository.php:912]:
```suggestion
            WHERE p.id = ? AND m.id IN (wl.movies_id) AND wl.user_id = ?
```

No need for this extra subquery

File: src/Domain/Movie/MovieRepository.php [hunk 2]
```diff
@@ -898,6 +898,42 @@ public function fetchWithDirector(int $personId, int $userId) : array
...
+    {
+        return $this->dbConnection->fetchAllAssociative(
+            <<<SQL
+            SELECT DISTINCT m.*, mur.rating as userRating
+            FROM movie m
+            JOIN movie_crew mc ON m.id = mc.movie_id AND job = "Director"
+            JOIN person p ON mc.person_id = p.id
+            JOIN watchlist wl ON m.id = wl.movie_id and wl.user_id = ?
+            LEFT JOIN movie_user_rating mur ON wl.movie_id = mur.movie_id and mur.user_id = ?
+            WHERE p.id = ? AND m.id IN (SELECT DISTINCT movie_id FROM watchlist)
```
Comment (leepeuker) [inline src/Domain/Movie/MovieRepository.php:930]:
```suggestion
            WHERE p.id = ? AND m.id IN (wl.movies_id)
```

same here

File: src/Domain/Movie/MovieRepository.php [hunk 3]
```diff
@@ -898,6 +898,42 @@ public function fetchWithDirector(int $personId, int $userId) : array
...
+            JOIN person p ON mc.person_id = p.id
+            JOIN watchlist wl ON m.id = wl.movie_id
+            LEFT JOIN movie_user_rating mur ON wl.movie_id = mur.movie_id and mur.user_id = ?
+            WHERE p.id = ? AND m.id IN (SELECT DISTINCT movie_id FROM watchlist) AND wl.user_id = ?
+            ORDER BY LOWER(m.title)
+            SQL,
+            [$userId, $personId, $userId],
+        );
+    }
+
```
Comment (leepeuker) [inline src/Domain/Movie/MovieRepository.php:918]:
```suggestion
```

File: templates/page/person.html.twig [hunk 4]
```diff
@@ -137,6 +137,27 @@
                         {% endfor %}
                     </div>
                 {% endif %}
+                {% if moviesFromWatchlistAsActor is not empty %}
+                    <h4>From your watch list</h4>
```
Comment (leepeuker) [inline templates/page/person.html.twig:141]:
```suggestion
                    <h6>From your watch list</h6>
```

Let's stay consistent with the headline sizes on the page, h4 seems to big to me anyway

File: templates/page/person.html.twig [hunk 5]
```diff
@@ -159,6 +180,28 @@
                         </div>
                     {% endfor %}
                 </div>
+
+                {% if moviesFromWatchlistAsDirector is not empty %}
+                  <h4>From your watch list</h4>
```
Comment (leepeuker) [inline templates/page/person.html.twig:185]:
```suggestion
                  <h6>From your watch list</h6>
```

**Comment by leepeuker** [comment]
Have a look at my comments please. Afterwards I will approve the PR. The layout will probably change again, maybe to something like I suggested [here](https://github.com/leepeuker/movary/pull/545#issuecomment-1806892234), but who knows when. So in the meantime this is a great solution to already see the watchlist movies :+1:

**Comment by sahinakkaya** [comment]
OK, done!

File: src/Domain/Movie/MovieRepository.php (hunk 1 above)
Comment (sahinakkaya) [reply src/Domain/Movie/MovieRepository.php]: I just tested this but it is giving errors: `[previous exception] [object] (Exception(code: 0): Unable to prepare statement: 1, no such column: wl.movies_id at /app/vendor/doctrine/dbal/src/Driver/SQLite3/Connection.php:25)
`

**Comment by sahinakkaya** [comment]
Converted it to draft because of https://github.com/leepeuker/movary/pull/545#discussion_r1390389897

File: src/Domain/Movie/MovieRepository.php (hunk 1 above)
Comment (leepeuker) [reply src/Domain/Movie/MovieRepository.php:912]: ups, I had a type, it is `movie_id` and not `movie_ids`, it should work now

**Review by leepeuker — APPROVED** [review]

**Comment by sahinakkaya** [comment]
Uhm... when is this going to be merged? I can't wait to use it in my server :)

**Pull request merged**